drivers: sensor: Update ICM45686 driver to use TDK HAL

JHKim8212 opened the pull request:
Subject : Update ICM45686 driver to use TDK HAL

Update : Driver Integration with TDK HAL. We have updated the driver to utilize the TDK HAL. The TDK HAL is implemented based on TDK InvenSense's eMD (Embedded Motion Driver) and significantly streamlines the driver development process.
Also APEX(Advanced Pedometer and Event Detection) features are applied.

Signed-off-by: JuHyun Kim <jh.kim@tdk.com>

JarmouniA reviewed (commented):
Hi, could you drop the merge commits, they aren't allowed.
Also keep in mind that the dev branch shall be buildable and runnable at each commit.

ubieda reviewed (commented):
Drive-by review:

- Please clean-up the PR addressing all compliance and SonarQube issues.
- Make sure to read through the docs for [Contributing to Zephyr](https://docs.zephyrproject.org/latest/contribute/index.html). For instance, don't use fix-up commits.
- Since this driver is being used by downstream users, may we split this PR in two patches?
    1. One making the changes on existing code to track regressions.
    2. One adding the new features (e.g: APEX).

Then, I'll put some time aside to test the PR with Sensor Streaming, which is my downstream usecase.

ubieda commented:
@JHKim8212 is this a conflict in the HAL or in the Zephyr drivers?

JHKim8212 commented:
@ubieda conflict in HAL.  
only need to load icm456xx_h for build this driver.
But twister loaded both of icm42x7x and icm456xx

-I/__w/zephyr/modules/hal/tdk/icm42x7x/icm42x7x_h 
-I/__w/zephyr/modules/hal/tdk/icm456xx 
-I/__w/zephyr/modules/hal/tdk/icm456xx/icm456xx_h *

ubieda commented:
> @ubieda conflict in HAL. only need to load icm456xx_h for build this driver. But twister loaded both of icm42x7x and icm456xx
> 
> -I/__w/zephyr/modules/hal/tdk/icm42x7x/icm42x7x_h -I/__w/zephyr/modules/hal/tdk/icm456xx -I/__w/zephyr/modules/hal/tdk/icm456xx/icm456xx_h *

I suggest submitting a fix to the HAL, then rebasing.

MaureenHelm commented:
Please fix the compliance and twister failures

rbuisson-invn committed:
modules: hal_tdk: Update for icm45686 device family

icm45686 is a 6-axis IMU sensor.
https://invensense.tdk.com/products/motion-tracking/6-axis/icm-45686/

Signed-off-by: Remi Buisson <remi.buisson@tdk.com>

rbuisson-invn committed:
drivers: sensor: icm45686: update drivers

Use official TDK Invensense driver for icm45686 sensor in tdk_hal module.
Supports icm456xx variants.

Signed-off-by: Remi Buisson <remi.buisson@tdk.com>

rbuisson-invn committed:
dts: bindings: sensor: Update invensense icm45686 properties

Add APEX related properties.
Add new icm456xx variants.

Signed-off-by: Remi Buisson <remi.buisson@tdk.com>
